src/backend/InvenTree/common/models.py

```diff
@@ -2926,7 +2959,7 @@ class Meta:
     @classmethod
     def check_recent(cls, key: str, uid: int, delta: timedelta):
         """Test if a particular notification has been sent in the specified time period."""
-        since = InvenTree.helpers.current_date() - delta
+        since = InvenTree.helpers.current_time() - delta
```

Comment:
Is this intentionally changed? This is breaking

Reply:
Is it? Did I miss some day-to-time conversions? As it stands, with current date, no notification will be sent twice within the same date. So I changed it to time, and only altered it if an override is supplied to the notification method.

Reply:
current_date returns a date object that is not aware of the specific time, current_time a datetime object so the behaviour changes and this is a public method (ie plugins like mine are using this already). The PR needs to be marked breaking and mentioned in the release note

Reply:
Hmm. That's definitely something. I could alter it so current_date is used if no delta override is defined? Then it won't be breaking?

Reply:
the problem is that a date is always 0hrs 0min while datetime is the real time so the value of since will be different

Reply:
I'll look into it. I'll implement it such that if you do not supply a delta, nothing's changed.

Reply:
Let me know if the revised solution is okay